src/config.rs

```diff
@@ -8,10 +8,59 @@ use serde_json::Value;
 use std::collections::HashMap;
 use std::mem;
 use std::str::FromStr;
+use std::sync::RwLock;
 use wildmatch::WildMatch;
+use lazy_static::lazy_static;
 
 const SETTINGS_JSON: &str = include_str!("./config/settings.json");
 
+struct GlobalStorage {
+    config_store: Option<ConfigStore>,
+}
+
+impl GlobalStorage {
+    pub fn config_store(&mut self) -> &mut ConfigStore {
+        match self.config_store {
+            Some(ref mut store) => store,
+            None => {
+                let store = ConfigStore::from_storage(Box::new(storage::memory_storage::MemoryStorageAdapter::new()), true).unwrap();
+                self.config_store = Some(store);
+                self.config_store.as_mut().unwrap()
+            }
+        }
+    }
+}
+
+lazy_static! {
+    static ref GLOBAL_STORAGE: RwLock<GlobalStorage> = RwLock::new(GlobalStorage {
+        config_store: None,
+    });
+}
+
+#[macro_export]
+macro_rules! config {
+    (string $key:expr) => {
+        let lock = GLOBAL_STORAGE.config_store().read();
+        let v = GLOBAL_STORAGE.config_store().get($key).to_string();
+        drop(lock);
+        v
+    };
+    (bool $key:expr) => {
+        GLOBAL_STORAGE.config_store().get($key).to_bool()
+    };
+    (uint $key:expr) => {
+        GLOBAL_STORAGE.config_store().get($key).to_uint()
+    };
+    (sint $key:expr) => {
+        GLOBAL_STORAGE.config_store().get($key).to_sint()
+    };
+    (map $key:expr) => {
+        GLOBAL_STORAGE.config_store().get($key).to_map()
+    };
+}
+
+unsafe impl Send for GlobalStorage {}
```

Comment:
This is telling the compiler:   nah, it's allright to use in multi threading...  

but i'm not sure that it is... we want to be able to lock in certain conditions (when writing the json file, or doing db writing)... so we have to fix that from the storage adapters..  when we know for sure that this is safe for multithreading, we somehow have to convince the compiler that this is all good...

we are working around the compiler here, so it might be tricky to get it right..

Reply:
To lock resources against shared access across different threads, you'd normally gate the resource behind an `Arc` / `Mutex`.  [Channels](https://doc.rust-lang.org/std/sync/mpsc/index.html) might also be useful here, depending on the specific thing being done.